eva-accession-core/src/test/java/uk/ac/ebi/eva/accession/core/mongoTemplate/MongoTemplateTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.ac.ebi.eva.accession.core.mongoTemplate;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.dao.DataAccessException;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.ac.ebi.eva.commons.core.models.AnnotationMetadata;
+
+import static org.mockito.Mockito.doThrow;
+
+@RunWith(SpringRunner.class)
+public class MongoTemplateTest {
+    @MockBean
+    private MongoTemplate mongoTemplate;
+
+    @Rule
+    public ExpectedException exceptionRule = ExpectedException.none();
+
+    @Before
+    public void setUp() {
+
+    }
+
+    @Test(expected = DataAccessException.class)
+    public void testMongoTemplateWriteResultChecking() {
+        AnnotationMetadata annotationMetadata = new AnnotationMetadata("vep_1", "vep_cache_1", true);
+        doThrow(new DataAccessException("Simulated exception") {})
```

Comment:
`DataAccessException` is getting thrown and caught by us. 
Is it not going through Spring ?

Reply:
Come to think of it, this test shouldn't even be there because it is tautologically true. But I guess it is only there as a placeholder if we **_eventually_** figure out a way to test the WriteResultChecking.

Reply:
To check **WriteResultChecking**, ideally we will need an integration test that can somehow throw an error when doing DB operations. As Sundar said, this is a just a placeholder for the time being.